test/storage/tile_compression_select_test.cpp

```diff
@@ -0,0 +1,80 @@
+#include <memory>
+#include <math.h>
+#include "sql/testing_sql_util.h"
+#include "catalog/catalog.h"
+#include "catalog/database_catalog.h"
+#include "storage/database.h"
+#include "storage/data_table.h"
+#include "concurrency/testing_transaction_util.h"
+#include "common/harness.h"
+#include "executor/create_executor.h"
+#include "planner/create_plan.h"
+#include "storage/tile_group_iterator.h"
+#include "storage/tile_group.h"
+#include "storage/dictionary_encoding_tile.h"
+
+namespace peloton {
+namespace test {
+class CompressionSelectTest : public PelotonTest {};
+TEST_F(CompressionSelectTest, BasicTest) {
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, txn);
+  catalog::Catalog::GetInstance()->Bootstrap();
+  txn_manager.CommitTransaction(txn);
+  // Create a t
+  txn = txn_manager.BeginTransaction();
+
+  TestingSQLUtil::ExecuteSQLQuery(
+      "CREATE TABLE foo(id integer, name varchar(32));");
+  std::string testTableName = "foo";
+
+	TestingSQLUtil::ExecuteSQLQuery("insert into foo values(1, 'taodai')");
+	TestingSQLUtil::ExecuteSQLQuery("insert into foo values(2, 'bohan')");
+	TestingSQLUtil::ExecuteSQLQuery("insert into foo values(3, 'siyuan')");
+	txn_manager.CommitTransaction(txn);
+	LOG_INFO("insert finish");
+	txn = txn_manager.BeginTransaction();
+	auto dataTable_object = catalog::Catalog::GetInstance()->GetTableWithName(DEFAULT_DB_NAME, DEFUALT_SCHEMA_NAME, testTableName, txn);
+	auto TGiterator = new storage::TileGroupIterator(dataTable_object);
```

Comment:
Why was this allocated on the heap? Just use a local var since its deleted 10 lines later...